lib/public_ip/service/json_type.rb

```diff
@@ -0,0 +1,22 @@
+require 'json'
+
+module PublicIp
+  module Service
+    class JsonType < Simple
+      extend PublicIp::Service::Registrable
+
+      # Default json parser
```

Comment:
I would rather not have any kind of implementation logic regarding parsing the responses other than calling the actual  parse method in the `ip` method below in these abstract service types.
It's the responsibility of the class that actually implements the service (the one that inherits this one) to deal with such details, what do you think?